community/values/src/main/java/org/neo4j/values/Value.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.values;
+
+public abstract class Value
+{
+    @Override
+    public abstract boolean equals( Object other );
```

Comment:
This method could probably be made final and implemented by delegation to the `equals( Value other )`-method. i.e. like this:

```
public final boolean equals( Object other ) {
    if ( this == other ) {
        return true;
    }
    return other instanceof Value && equals( (Value)other );
}
```